### PR TITLE
Add key usage extensions to certificates

### PR DESCRIPTION
**Pull request by mrohera**
Key usage determines what the certificate keys should be able to perform. More about this can be found here: https://www.ibm.com/support/knowledgecenter/en/SSKTMJ_9.0.1/admin/conf_keyusageextensionsandextendedkeyusage_r.html.

By adding these usage extensions, we further harden the Edge related certificate generation.

### CODE REVIEW

File: edgelet/hsm-sys/azure-iot-hsm-c/src/edge_pki_openssl.c [hunk 1]
```diff
@@ -795,6 +799,78 @@ static int set_basic_constraints(X509 *x509_cert, CERTIFICATE_TYPE cert_type, in
     return result;
 }
 
+static int add_ext(X509 *x509_cert, int nid, char *value, const char* nid_diagnostic)
```
Comment (myagley) [inline edgelet/hsm-sys/azure-iot-hsm-c/src/edge_pki_openssl.c]: Are we using this call yet? Or is this in prep for the identity cert work?

Comment (mrohera) [reply edgelet/hsm-sys/azure-iot-hsm-c/src/edge_pki_openssl.c]: Yes we are starting with this PR and this is also prep for ID cert coming up next.

Comment (mrohera) [reply edgelet/hsm-sys/azure-iot-hsm-c/src/edge_pki_openssl.c]: This also solves the problem where the quick start certs could not be used for E2E TG tests.

Comment (myagley) [reply edgelet/hsm-sys/azure-iot-hsm-c/src/edge_pki_openssl.c]: ok cool

**Pull request merged**